packages/playwright-core/src/client/fetch.ts

```diff
@@ -47,7 +47,7 @@ export type FetchOptions = {
 
 type NewContextOptions = Omit<channels.PlaywrightNewRequestOptions, 'extraHTTPHeaders' | 'clientCertificates' | 'storageState' | 'tracesDir'> & {
   extraHTTPHeaders?: Headers,
-  storageState?: string | StorageState,
+  storageState?: string | SetStorageState,
```

Comment:
This change seems unrelated, please move it into some IndexedDB pr instead?

Reply:
It's related! `NewContextOptions` is for writing, so it needs to use `SetStorageState`. This should've been fixed earlier, but was always covered because we had all these `(... as any)`s in the code.